Introduce SOE content

asteflova opened the pull request:
_I'm trying to find a way to introduce the [SOE content](https://github.com/theforeman/foreman-documentation/commit/4d99fe84535ab38bf04dc823b33bdaede4fc9b12) to the upstream docs set in a way that's useful, maintainable, and doesn't duplicate existing content. This PR is a WIP effort to 'kind of' finalize the first of the 10 steps of the SOE guide and to figure out a way to configure a preview._

The SOE guide is a few years old and heavily outdated. The goal is to update it, modularize it, and publish upstream. But there are caveats:

* “Just” copy-pasting the contents of the SOE guide and updating it with the latest functionality would introduce a lot of duplication into the docs set. A lot of the topics in the SOE guide are already covered by existing modules.
* Our current docs set provides a rich repository of modules on various useful topics; _but finding the right modules at the right times might sometimes be difficult_ – and this is something a new guide based on SOE could solve. For example:
  * SOE Step One covers setting up DNS, logging in to web UI, using Hammer CLI, an overview of organizations + how to create a custom organization – all in one chapter, in sequence.
  * In the existing docs set, these topics are covered over 3 different guides and the respective modules are included among a huge number of other modules so it’s not easy for users to find them.

The benefit of bringing SOE into the fold could be to provide a more focused document that would be designed to guide users through the process of deploying Foreman after the initial installation, but would not cover all the features of a given topic. A guide that is:
* Very user-focused
* Based on scenarios
* Limited to the most common workflows (linking to the less common workflows when needed)
* Explaining the when and the why

Contents of this PR:

* master.adoc for a new guide (proposals for a good title would be appreciated)
* A list of assemblies (created specifically for SOE) that include several existing modules (pre-edited in previous PRs to fit into SOE)
* Overall: The first step of a curated content path that users can follow to set up their environment without having to sift through large guides.

Target audience: Users who have recently installed a Foreman server and are looking for further steps on how to further develop their infrastructure.
* What they want: Easy-to-follow steps and examples (“So I’ve just installed a Foreman server, and now what?” or “So I can see that Foreman supports creating organizations, but how can I make the most out of them?”) with links to additional deep-dive docs that they can follow if they choose to.
* What they don’t want: To be overwhelmed with too much information at this point (either in the form of heavy-to-read conceptual modules or a vast repertoire of all the configuration options Foreman offers).

* [ ] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (planned Satellite 6.15)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

ekohl commented:
![image](https://github.com/theforeman/foreman-documentation/assets/155810/fc8b7369-1740-4147-b6c6-cee2241b9ea1)

asteflova commented:
I know, I realized right after opening this PR that introducing it with "Nothing to see here" is probably the single surest way to make *everyone* take a look (which is good to know actually, I'll keep it in mind for future PRs). I updated the description to provide more details while I fumble on.

ekohl commented:
To be fair, I didn't look at the content. Just had to respond to the South Park reference

ekohl commented on guides/doc-Creating_a_Standard_Operating_Environment/master.adoc:
```diff
@@ -0,0 +1,28 @@
+
+= {SOEDocTitle}
+
+// This guide is not ready for stable releases
+ifdef::HideDocumentOnStable[]
+include::common/modules/snip_guide-not-ready.adoc[]
+endif::[]
+ifndef::HideDocumentOnStable[]
```
Would it make sense to start developing this guide in a way that it is suitable for stable releases? Looks like you're not requesting cherry picks, so that means we only need to make sure the content is OK even after branching.

asteflova replied:
To be honest, I only copy-pasted all this metadata from another guide without really giving it much thought at this point. The next step for this PR is to discuss it with the rest of the team (e.g. does it even make sense to add the SOE content in this way?). If the team agrees that this effort is worth continuing, I'll think about how and where to start publishing it.

apinnick replied:
Considering the size of the SOE guide, I am concerned that the PRs will become unmanageable. Suggest you start with some kind of 'SOE planning' doc (Google doc or miro board) so that you can move sections around and add comments before you make a final decision on how to organize the guide.

asteflova replied:
@apinnick So far, my approach has been to submit any changes to the existing content in smaller PRs (https://github.com/theforeman/foreman-documentation/pull/2847, https://github.com/theforeman/foreman-documentation/pull/2846, https://github.com/theforeman/foreman-documentation/pull/2861), and use this PR to create new SOE-specific files and only minor additional changes. That should prevent this PR from becoming unamangeable. I hope.

Also, as for other tools to plan this, I've tried them all: google doc, miro, a spreadsheet... Miro can't be shared upstream and a spreadsheet turned out to be a lot less useful than a GH preview.

apinnick replied:
Pity about Miro! Drag-and-drop is a lot easier than cut/paste. :-(

maximiliankolb replied:
I think we should not move away from this PR. It's totally OK to apply major concerns now; and then rework the structure in a follow-up PR and fix other minor issues related to wording/style etc.

In the two Google docs for doc PoCs that I have access to, there has not been any discussion in March at all; so I conclude that folks are more active/comfortable on GitHub.

asteflova replied:
> Pity about Miro! Drag-and-drop is a lot easier than cut/paste. :-(

@apinnick On another upstream project (Ansible, I think?), they have good experience with https://www.mindmeister.com/ for the purpose of defining user stories. That does have drag-and-drop functionality. Working through PRs seems to work quite well without the overhead of additional tools, but I could give mindmeister a try too if this team would appreciate it. Let me know.

apinnick replied:
Sounds interesting! I will check it out. Thanks for the link.

Lennonka commented on guides/common/attributes-titles.adoc:
```diff
@@ -28,6 +28,7 @@
 :UpgradingDocTitle: Upgrading {ProjectName} to {ProjectVersion}
 :UpgradingDisconnectedDocTitle: Upgrading Disconnected {ProjectName} to {ProjectVersion}
 :UpgradingPreviousDocTitle: Upgrading {ProjectName} to {ProjectVersionPrevious}
+:SOEDocTitle: Creating a standard operating environment with {ProjectName}
```
Order alphabetically?

Lennonka commented on guides/doc-Creating_a_Standard_Operating_Environment/docinfo.html:
Thi file shouldn't be needed.

Lennonka reviewed (changes requested):
This #2872 seems to be messing with the build. After this PR is merged, rebase your PR and it should build the preview.

Lennonka commented on guides/doc-Creating_a_Standard_Operating_Environment/master.adoc:
```diff
@@ -0,0 +1,28 @@
+// Step One
+include::common/modules/con_setting-up-your-project-system-management-infrastructure.adoc[leveloffset=+1]
+
+include::common/assembly_configuring-dns-dhcp-and-tftp.adoc[leveloffset=+1]
```
```suggestion
:ProductName: {ProjectServer}
include::common/assembly_configuring-dns-dhcp-and-tftp.adoc[leveloffset=+1]

:ProductName: {SmartProxyServer}
include::common/assembly_configuring-dns-dhcp-and-tftp.adoc[leveloffset=+1]
```
Probaby like this?
In any case, you're missing a `:ProductName:` definition for this module and that's why the build fails.

asteflova commented:
I updated the PR description with (a lot) more details about what I'm hoping to accomplish. I'd also like to bring this up at the next [community meeting](https://community.theforeman.org/t/documentation-team-meeting-2024-03-21/37166) to ask for general feedback about the direction I'm taking.

maximiliankolb commented:
I had a quick look at https://theforeman-foreman-documentation-preview-pr-2866.surge.sh/nightly/Creating_a_Standard_Operating_Environment/index-katello.html and I'm still a bit confused. Is there any reasoning to this order yet or is that just a draft?

* still mentions SCA which should get dropped for nightly
* some graphics, maybe a better version of https://docs.orcharhino.com/or/docs/sources/introduction/first_steps.html, would really help
* some introduction on _why_ to think about certain areas such as DHCP/DNS/TFTP and eventually configure them

I believe Foreman docs will benefit greatly from a guide that tells users "look, we have like 1001 options to configure, but to get you started, have a look at those 10. here's what you probably want to take care of before deploying your first host/providing content to the first host". So :+1: on your effort.

asteflova commented:
> I had a quick look at https://theforeman-foreman-documentation-preview-pr-2866.surge.sh/nightly/Creating_a_Standard_Operating_Environment/index-katello.html and I'm still a bit confused. Is there any reasoning to this order yet or is that just a draft?

Thanks for taking a look! The order follows the structure of the original SOE guide. You can view the content upstream under the link in this PR's description; for Step 1 specifically, [here](https://github.com/theforeman/foreman-documentation/commit/4d99fe84535ab38bf04dc823b33bdaede4fc9b12#diff-76eede0b5db0ca832202e8cff7cbfac757a9cfd1f9bdaa561489864a6b985e3d). I left out some pieces that seem out of scope of Foreman docs (like recommendations for version control tools).

asteflova commented:
Max and I talked about the purpose of the new guide. As a result of that conversation, I'll be adding modules on recommendations/best practices/examples; for now, only as placeholders to be populated later. These best practices could eventually look like this:
* https://docs.orcharhino.com/or/docs/sources/guides/alma_linux/managing_content/importing_content.html#Best_Practices_for_Products_and_Repositories
* https://docs.orcharhino.com/or/docs/sources/guides/alma_linux/managing_content/managing_activation_keys.html#Best_Practices_for_Activation_Keys

ekohl commented on guides/doc-Creating_a_Standard_Operating_Environment/master.adoc:
```diff
@@ -0,0 +1,29 @@
+
+include::common/assembly_configuring-dns-dhcp-and-tftp.adoc[leveloffset=+1]
+
+include::common/assembly_enabling-and-synchronizing-the-project-client-name-repository.adoc[leveloffset=+1]
```
If you want this guide to make sense for non-Katello, this should be hidden.
```suggestion
ifdef::katello,satellite,orcharhino[]
include::common/assembly_enabling-and-synchronizing-the-project-client-name-repository.adoc[leveloffset=+1]
endif::[]
```

asteflova replied:
Addressed... by removing the whole assembly! It's currently part of the installation guide. For SOE, I want to try to focus on post-installation guide tasks only. At least for now, until I change my mind. Again.

ekohl commented on guides/common/assembly_administering-project-with-projectwebui.adoc:
```diff
@@ -0,0 +1,17 @@
+
+Get to know the basics of administering {Project} from the {ProjectWebUI}.
+
+include::modules/proc_importing-the-katello-root-ca-certificate.adoc[leveloffset=+1]
```
I'm wondering about this. This is Katello specific, so it should be hidden. We right now hide it like this:
https://github.com/theforeman/foreman-documentation/blob/19b38e6be136208a7b43e8d5bf8c92a8d1f0a682/guides/common/assembly_accessing-server.adoc#L5-L7

But would it make more sense to include those ifdefs in the procedure itself to avoid duplicating it like this:

```suggestion
ifdef::katello,orcharhino,satellite[]
include::modules/proc_importing-the-katello-root-ca-certificate.adoc[leveloffset=+1]
endif::[]
```

asteflova replied:
I created https://github.com/theforeman/foreman-documentation/pull/3090 for this. Let's see if the team agrees.

asteflova replied:
Based on https://github.com/theforeman/foreman-documentation/pull/3090, let's stick to including ifdefs in parent files rather than in the included files themselves.

asteflova replied:
Added.

ekohl reviewed (commented):
Like @maximiliankolb I'm trying to figure out the structure. My first thought is that starting with the DNS setup is really throwing you into the deep end. Is that really the start? Then you synchronize content using Hammer, but only give the introduction to Hammer later.

asteflova commented:
First of all, thanks a lot for letting me know what confuses you about this @ekohl @maximiliankolb. I've been dealing with the original SOE guide for so long now that I don't have a fresh perspective anymore and don't see these gaps.

I think that the structure defined by the SOE guide's ToC must make _some_ sense, otherwise our customer-facing teams wouldn't have written it. But I'm all for improving it whenever needed.

This is more of a long-term goal for me, but I'll keep addressing your feedback in the coming weeks. My intentions is to get Step 1 in a good-enough shape to merge (but not publish just yet -- just as was the case with the previous attempt in https://github.com/theforeman/foreman-documentation/pull/555 ).

ekohl commented:
Today I talked with @apinnick and we discussed the [Satellite Quick Start Guide](https://access.redhat.com/documentation/en-us/red_hat_satellite/6.10/html/quick_start_guide/index) that was published in the past.

I actually like the structure, but just reintroducing it now might mean we have a lot of overlap. There's the installation guide, which covers a lot of this already but in the SOE guide I also see a lot of these topics.

Would it make sense to take the Quick Start Guide structure to start with and start filling in the areas?

maximiliankolb commented on guides/common/assembly_administering-project-with-hammer.adoc:
```diff
@@ -0,0 +1,15 @@
+
+include::modules/con_hammer-cli-overview.adoc[leveloffset=+1]
+
+== Additional resources
+
```
```suggestion
.Additional resources
```

alternative: move to separate "snip_" file. But by convension, we should not have any heading/text in assemblies.

maximiliankolb commented on guides/common/assembly_administering-project-with-hammer.adoc:
```diff
@@ -0,0 +1,15 @@
+[id="administering-project-with-hammer_{context}"]
+= Getting started with using Hammer to manage {Project}
+
+Get to know the basics of administering {Project} from the command line by using the Hammer tool.
```
Please move to a "con_" file.

maximiliankolb commented on guides/common/assembly_administering-project-with-projectwebui.adoc:
```diff
@@ -0,0 +1,17 @@
+[id="administering-project-with-projectwebui_{context}"]
+= Getting started with {ProjectWebUI}
+
+Get to know the basics of administering {Project} from the {ProjectWebUI}.
```
Please move to a "con_" file.

maximiliankolb commented on guides/common/assembly_deploying-organizations-in-project.adoc:
```diff
@@ -0,0 +1,11 @@
+[id="deploying-organizations-in-project_{context}"]
+= Deploying organizations in {Project}
+
+{Project} uses organizations to divide hosts into logical groups based on ownership, purpose, content, security level, or other criteria.
+You can view, create, and manage {Project} organizations by using {ProjectWebUI} or the Hammer CLI tool.
```
Please move to a "con_" file.

asteflova commented:
> Would it make sense to take the Quick Start Guide structure to start with and start filling in the areas?

One thing that concerns me is that the Quickstart Guide's structure does not match the SOE guide's structure at all:
* https://docs.redhat.com/en/documentation/red_hat_satellite/6.10/html-single/quick_start_guide/index#creating_new_domain
* https://access.redhat.com/articles/1585273

And what I'm being asked to do downstream is to work with the SOE guide because that's built on the experience of our customer-facing teams. Which is always a good thing.

maximiliankolb commented:
> Would it make sense to take the Quick Start Guide structure to start with and start filling in the areas?

Does this relate to [Basic content management workflow](https://docs.theforeman.org/nightly/Managing_Content/index-katello.html#Basic_Content_Management_Workflow_content-management)? I like the idea for upstream and downstream; it's currently upstream-only; in reality, it makes the already big guide for Katello just bigger.

Lennonka commented on guides/doc-Creating_a_Standard_Operating_Environment/master.adoc:
```diff
@@ -0,0 +1,18 @@
+include::common/modules/proc_providing-feedback-on-red-hat-documentation.adoc[leveloffset=+1]
+endif::[]
+
+include::common/assembly_administering-project-with-projectwebui.adoc[leveloffset=+1]
+
+include::common/assembly_administering-project-with-hammer.adoc[leveloffset=+1]
```
How about making this into a single chapter, e.g. "Interfacing with Project", and including API and webhooks? As in "what ways there are to get data in and out?".

asteflova commented:
@Lennonka suggested that an alternative way of capturing the [SOE](https://github.com/theforeman/foreman-documentation/commit/4d99fe84535ab38bf04dc823b33bdaede4fc9b12) content flow into the docs set would be to use `.Next steps` sections to guide the users through the SOE steps. I think that approach is worth exploring.

Benefit: We wouldn't need to introduce yet another guide to the existing portfolio.

Risk: Adding links and forcing users to open new and new tabs might not be the best user experience.

maximiliankolb commented:
Also something potentially for SOE: `guides/common/modules/ref_initialization-script-for-provisioning-examples.adoc` (maybe not in _this_ version).

asteflova commented:
Just a note: I opened another PR that's part of the SOE effort: https://github.com/theforeman/foreman-documentation/pull/3281 which is related to Step 2 of the original SOE guide.

asteflova commented:
https://github.com/theforeman/foreman-documentation/pull/3456 effectively replaces this PR. On our downstream side, noone has brought up the need for a separate guide in a long time. And https://github.com/theforeman/foreman-documentation/pull/3456 can most likely get the job done just as well and can be finalized and reviewed a lot faster.
